plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergModule.java

```diff
@@ -96,4 +100,12 @@ public void configure(Binder binder)
         tableProcedures.addBinding().toProvider(ExpireSnapshotsTableProcedure.class).in(Scopes.SINGLETON);
         tableProcedures.addBinding().toProvider(RemoveOrphanFilesTableProcedure.class).in(Scopes.SINGLETON);
     }
+
+    @Singleton
+    @Provides
+    @ForMetadataFetching
+    public ForkJoinPool createIcebergMetadataFetchingForkJoinPool(IcebergConfig icebergConfig)
+    {
+        return new ForkJoinPool(icebergConfig.getMaxConcurrentMetadataLoaders());
```

Comment:
who shuts this executor down on connector shutdown?

Reply:
I don't have a better idea that what I tried here. The couple executor services in the Hive connector are never shut down from what I can tell. Got any better ideas?